google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryAggregationsTest.java

```diff
@@ -0,0 +1,856 @@
+        map(
+            "a",
+                map(
+                    "author",
+                    "authorA",
+                    "title",
+                    "titleA",
+                    "pages",
+                    100,
+                    "year",
+                    1980,
+                    "rating",
+                    asList(5, 1000)),
+            "b",
+                map(
+                    "author", "authorB", "title", "titleB", "pages", 50, "year", 2020, "rating",
+                    asList(4)),
+            "c",
+                map(
+                    "author",
+                    "authorC",
+                    "title",
+                    "titleC",
+                    "pages",
+                    100,
+                    "year",
+                    1980,
+                    "rating",
+                    asList(2222, 3)),
+            "d",
+                map(
+                    "author", "authorD", "title", "titleD", "pages", 50, "year", 2020, "rating",
+                    asList(0)));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot =
+        collection
+            .whereArrayContainsAny("rating", asList(5, 3))
+            .aggregate(
+                sum("rating"),
+                average("rating"),
+                sum("pages"),
+                average("pages"),
+                AggregateField.count())
+            .get()
+            .get();
+    assertThat(snapshot.get(sum("rating"))).isEqualTo(0);
+    assertThat(snapshot.get(average("rating"))).isEqualTo(null);
+    assertThat(snapshot.get(sum("pages"))).isEqualTo(200);
+    assertThat(snapshot.get(average("pages"))).isEqualTo(100);
+    assertThat(snapshot.get(AggregateField.count())).isEqualTo(2);
+  }
+
+  @Test
+  public void performsAggregationsOnNestedMapValues() throws Exception {
+    assumeTrue(
+        "Skip this test when running against prod because it requires composite index creation.",
+        isRunningAgainstFirestoreEmulator(firestore));
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "a",
+            map(
+                "author",
+                "authorA",
+                "title",
+                "titleA",
+                "metadata",
+                map("pages", 100, "rating", map("critic", 2, "user", 5))),
+            "b",
+            map(
+                "author",
+                "authorB",
+                "title",
+                "titleB",
+                "metadata",
+                map("pages", 50, "rating", map("critic", 4, "user", 4))));
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+    AggregateQuerySnapshot snapshot =
+        collection
+            .aggregate(
+                sum("metadata.pages"),
+                average("metadata.pages"),
+                average("metadata.rating.critic"),
+                sum("metadata.rating.user"),
```

Comment:
nit: This test could run against prod and provide value by only running against "metadata.pages".

Reply:
Fixed.